diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -107,56 +125,127 @@ class FileStructureRule(private val configRules: List<RulesConfig>) : Rule("file
+                        if (index != group.size - 1) {
+                            node.addChild(PsiWhiteSpaceImpl("\n"), null)
+                        }
+                    }
+                    if (groupIndex != size - 1) {
+                        node.addChild(PsiWhiteSpaceImpl("\n\n"), null)
+                    }
+                }
             }
-            node.addChild(importNode, null)
-        }
     }
 
-    private fun ASTNode.getSiblingBlocks(copyrightComment: ASTNode?,
-                                         headerKdoc: ASTNode?,
-                                         fileAnnotations: ASTNode?,
-                                         packageDirectiveNode: ASTNode
+    private fun ASTNode.getSiblingBlocks(
+        copyrightComment: ASTNode?,
+        headerKdoc: ASTNode?,
+        fileAnnotations: ASTNode?,
+        packageDirectiveNode: ASTNode
     ): Pair<ASTNode?, ASTNode> = when (elementType) {
         BLOCK_COMMENT -> null to listOfNotNull(headerKdoc, fileAnnotations, packageDirectiveNode).first()
         KDOC -> copyrightComment to (fileAnnotations ?: packageDirectiveNode)
         FILE_ANNOTATION_LIST -> (headerKdoc ?: copyrightComment) to packageDirectiveNode
         else -> error("Only BLOCK_COMMENT, KDOC and FILE_ANNOTATION_LIST are valid inputs.")
     }
 
+    private data class ImportGroups(
+        val android: List<KtImportDirective>,
+        val ownDomain: List<KtImportDirective>,
+        val others: List<KtImportDirective>,
+        val javaCore: List<KtImportDirective>,
+        val kotlinCore: List<KtImportDirective>
+    ) {
+        fun toList() = listOf(android, ownDomain, others, javaCore, kotlinCore)
+
+        companion object {
+            fun create(imports: List<KtImportDirective>, domainName: String): ImportGroups {
+                val (android, notAndroid) = imports.partitionByImportParts {
+                    it.isStandardAndroid()
+                }
+
+                val (ownDomain, tmp) = notAndroid.partitionByImportParts { paths ->
+                    paths.map { it.identifier }.zip(domainName.split(PACKAGE_SEPARATOR)).all { it.first == it.second }
+                }
+
+                val (others, javaAndKotlin) = tmp.partitionByImportParts {
+                    !it.isStandardJava() && !it.isStandardKotlin()
+                }
+
+                val (java, kotlin) = javaAndKotlin.partitionByImportParts { it.isStandardJava() }
+
+                return ImportGroups(android, ownDomain, others, java, kotlin)
+            }
+
+            @Suppress("UnsafeCallOnNullableType")
+            private inline fun List<KtImportDirective>.partitionByImportParts(predicate: (List<Name>) -> Boolean) =
+                partition {
+                    it.importPath!!.fqName.pathSegments().let(predicate)
+                }
+
+            private fun List<Name>.isStandardAndroid() =
+                first().identifier.let { it == "android" || it == "androidx" } ||
```

Comment:
oh, need to move these to constants

Reply:
I'd better extract lists of strings to `ASTNodeConstants`.